src/main/java/com/example/codebase/domain/notification/service/NotificationEventService.java

```diff
@@ -0,0 +1,149 @@
+    private static final long REDIS_EXPIRE_DURATION = 1000 * 60 * 30;
+
+    public NotificationEventService(NotificationReceivedStatusRepository notificationReceivedStatusRepository, RedisUtil redisUtil) {
+        this.notificationReceivedStatusRepository = notificationReceivedStatusRepository;
+        this.redisUtil = redisUtil;
+    }
+
+    @Transactional
+    public SseEmitter connection(Member member) {
+        int count = notificationReceivedStatusRepository.countByMember(member);
+        NotificationResponse.EventMessage eventMessage = NotificationResponse.EventMessage.from(null, count, null, LocalDateTime.now().toString());
+
+        registerUserConnection(member, eventMessage);
+
+        SseEmitter emitter = new SseEmitter(SSE_SESSION_TIMEOUT);
+
+        emitter.onTimeout(() -> {
+            log.info("onTimeOut callback");
+            emitter.complete(); // 시간 초과 발생시 -> Complete 호출
+        });
+
+        emitter.onCompletion(() -> {
+            log.info("onCompletion callback");
+            userEmitters.remove(member.getUsername());
+        });
+
+        userEmitters.put(member.getUsername(), emitter);
+
+        try {
+            emitter.send(SseEmitter.event().name("notificaiton_connection").data(eventMessage));
+            log.info("SSE 연결 성공: {}", member.getUsername());
+            return emitter;
+        } catch (IOException e) {
+            emitter.completeWithError(e);
+            return emitter;
+        }
+    }
+
+    private void registerUserConnection(Member member, NotificationResponse.EventMessage eventMessage) {
+        String key = getNotificationKey(member);
+        redisUtil.getData(key).ifPresentOrElse(json -> {
+               // JSON이 존재하는 경우, 업데이트
+                int count = notificationReceivedStatusRepository.countByMember(member);
+                eventMessage.setCount(count);
+                redisUtil.saveNotificationMessage(key, eventMessage, REDIS_EXPIRE_DURATION);
+        }, () -> {
+            // JSON이 존재하지 않는 경우, 저장
+            redisUtil.saveNotificationMessage(key, eventMessage, REDIS_EXPIRE_DURATION);
+        });
+    }
+
+    private void sendSseEvent(String username, NotificationResponse.EventMessage eventMessage) {
+        SseEmitter emitter = userEmitters.get(username);
+        if (emitter != null) {
+            try {
+                emitter.send(SseEmitter.event().name("notification").data(eventMessage));
+                log.info("SSE 알림 전송: {}", username);
+            } catch (IOException e) {
+                emitter.completeWithError(e);
+            }
+        }
+    }
+
+    public void disconnection(Member member) {
+        userEmitters.remove(getNotificationKey(member));
+        redisUtil.deleteData(getNotificationKey(member));
+    }
+
+    private static String getNotificationKey(Member member) {
+        return member.getUsername() + "_notification";
+    }
+
+
+    @Transactional
+    public void upDateCountAndSend(Member member, int count) {
```

Comment:
sendNotificationCount()메소드명이 로직상 의미가 맞는것 같아요!

```suggestion
    public void sendNotificationCount(Member member, int count) {
```